test/functional/cinder_controller_test.go

```diff
@@ -863,6 +863,75 @@ var _ = Describe("Cinder controller", func() {
 
 	})
 
+	When("Cinder CR instance is built with ExtraMounts", func() {
+		BeforeEach(func() {
+			rawSpec := map[string]interface{}{
+				"secret":              SecretName,
+				"databaseInstance":    "openstack",
+				"rabbitMqClusterName": "rabbitmq",
+				"extraMounts":         GetExtraMounts(),
+				"cinderAPI": map[string]interface{}{
+					"containerImage": cinderv1.CinderAPIContainerImage,
+				},
+				"cinderScheduler": map[string]interface{}{
+					"containerImage": cinderv1.CinderSchedulerContainerImage,
+				},
+				"cinderVolumes": map[string]interface{}{
+					"volume1": map[string]interface{}{
+						"containerImage": cinderv1.CinderVolumeContainerImage,
+					},
+				},
+			}
+
+			DeferCleanup(th.DeleteInstance, CreateCinder(cinderTest.Instance, rawSpec))
+			DeferCleanup(k8sClient.Delete, ctx, CreateCinderMessageBusSecret(cinderTest.Instance.Namespace, cinderTest.RabbitmqSecretName))
+			DeferCleanup(
+				mariadb.DeleteDBService,
+				mariadb.CreateDBService(
+					cinderTest.Instance.Namespace,
+					GetCinder(cinderTest.Instance).Spec.DatabaseInstance,
+					corev1.ServiceSpec{
+						Ports: []corev1.ServicePort{{Port: 3306}},
+					},
+				),
+			)
+			infra.SimulateTransportURLReady(cinderTest.CinderTransportURL)
+			DeferCleanup(infra.DeleteMemcached, infra.CreateMemcached(namespace, cinderTest.MemcachedInstance, memcachedSpec))
+			infra.SimulateMemcachedReady(cinderTest.CinderMemcached)
+			keystoneAPIName := keystone.CreateKeystoneAPI(cinderTest.Instance.Namespace)
+			DeferCleanup(keystone.DeleteKeystoneAPI, keystoneAPIName)
+			mariadb.SimulateMariaDBAccountCompleted(cinderTest.Database)
+			mariadb.SimulateMariaDBDatabaseCompleted(cinderTest.Database)
+			th.SimulateJobSuccess(cinderTest.CinderDBSync)
+			keystone.SimulateKeystoneEndpointReady(cinderTest.CinderKeystoneEndpoint)
+		})
+
+		It("Check the extraMounts of the resulting StatefulSets", func() {
+			th.SimulateStatefulSetReplicaReady(cinderTest.CinderAPI)
+			th.SimulateStatefulSetReplicaReady(cinderTest.CinderScheduler)
+			// Retrieve the generated resources
+			volume := cinderTest.CinderVolumes[0]
+			th.SimulateStatefulSetReplicaReady(volume)
+			ss := th.GetStatefulSet(volume)
+			// Check the resulting deployment fields
+			Expect(int(*ss.Spec.Replicas)).To(Equal(1))
+			Expect(ss.Spec.Template.Spec.Volumes).To(HaveLen(15))
+			Expect(ss.Spec.Template.Spec.Containers).To(HaveLen(2))
+			// Get the manila-share container
+			container := ss.Spec.Template.Spec.Containers[1]
+			// Fail if manila-share doesn't have the right number of
+			// VolumeMounts entries
+			Expect(container.VolumeMounts).To(HaveLen(17))
```

Comment:
we have https://github.com/openstack-k8s-operators/lib-common/blob/main/modules/common/test/helpers/volumes.go#L36 to check that an expected VolumeMount exist, but it does not (yet) provide functionality to validate the `MountPath`. should we enhance it?

Reply:
This seems a good idea as well. I was planning, as part of the other comment, to add a validation  as part of the `extraMount.propagation`, to make sure the `VolumeMount` exists in the resulting statefulset, but perhaps when you mentioned `validate the MountPath` you had something different in mind (e.g. reserved paths or other relevant `RFCs`)

Reply:
in your previous version you were validating the volumemount path like

~~~
Expect(vm.MountPath).To(	ContainSubstring(CinderCephExtraMountsPath))
~~~

we could enhance the test helpers to be able to do that. thats not something it checks.

Reply:
oh I see sorry. Even for `MountPath` we currently check the name and it would be good to be able to validate the path that way. I will add it to the existing assertion in `lib-common`, so we get it here with the next bump.